presto-hive/src/main/java/com/facebook/presto/hive/security/SqlStandardAccessControl.java

```diff
@@ -376,6 +377,20 @@ public void checkCanShowGrants(ConnectorTransactionHandle transaction, Identity
         return grantInfos;
     }
 
+    @Override
+    public void checkCanShowRoles(ConnectorTransactionHandle transactionHandle, Identity identity, String catalogName)
```

Comment:
@arhimondr back to the original question, we need both because you can't throw an exception in filter b/c that's used by information schema. All show access control has both canCheck and filter(). In practice, we actually don't use filter b/c only admins can even see the roles table in information_schema, but this is important for future extension/other connectors.